osu.Game/Users/UserActivity.cs

```diff
@@ -14,6 +15,8 @@ public abstract class UserActivity
         public abstract string Status { get; }
         public virtual Color4 GetAppropriateColour(OsuColour colours) => colours.GreenDarker;
 
+        public virtual string GetDetails(DiscordRichPresenceMode privacyMode) => string.Empty;
```

Comment:
Method needs to have xmldoc explaining its purpose.

I also dislike that the usage of the "rich presence mode" argument hard-couples user activity (which is not really discord-specific) with a discord-specific enumeration type. Especially so that the implementations of this method return non-empty contents for `DiscordRichPresenceMode.Off`, which looks like a bug.

Reply:
this is a fair point- I originally moved `getDetails` into `UserActivity` because it felt like a valid opportunity to make use of inheritance, but I also now think that it's probably better suited to live in `DiscordRichPresence`